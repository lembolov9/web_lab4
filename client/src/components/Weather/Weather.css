.Weather {
    margin: 20px;
    background-color: thistle;
    padding: 10px;
    border-radius: 15px;
}

.flex {
    display: flex;
    flex-wrap: wrap;
}

.grid {
    display: grid;
    grid-template-columns: 1fr 1fr;
}

.grid > .grid {
    grid-template-rows: 0.3fr 1fr ;
}

.grid > div {
    margin: 0 20px;
}

.grid > .cityTemp {
    font-size: 3.5em;
}

.grid > .Info {
    justify-content: space-between;
}

.cityName {
    font-family: Ubuntu, sans-serif;
    grid-column-start: 1;
    grid-column-end: 2;
    grid-row-start: 1;
    grid-row-end: 1;
}
.cityIcon {
    grid-column-start: 1;
    grid-column-end: 1;
    grid-row-start: 2;
    grid-row-end: 2;
}
.cityTemp {
    grid-column-start: 2;
    grid-column-end: 2;
    grid-row-start: 2;
    grid-row-end: 2;
}


.Info {
    flex-direction: column;
}


.propsRow {
    justify-content: space-between;
}

.mainInfo > div {
    font-weight: bold;
    margin: auto 0;
}

.deleteBtn {
    border-radius: 50%;
    transform: rotate(-45deg);
}

.loader {
    margin: auto;
    width: 6em;
    height: 6em;
    border: 1.1em solid rgba(0, 0, 0, 0.2);
    border-left: 1.1em solid lavender;
    border-radius: 50%;
    animation: load8 1.1s infinite linear;
    transition: opacity 0.3s;
}

@keyframes load8 {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(360deg);
    }
}