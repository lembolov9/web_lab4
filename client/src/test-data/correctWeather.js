export default  {
    "coord": {
        "lon": 30.52,
        "lat": 50.43
    },
    "weather": [
        {
            "id": 500,
            "main": "Rain",
            "description": "light rain",
            "icon": "10n"
        }
    ],
    "base": "stations",
    "main": {
        "temp": 8.02,
        "pressure": 1011,
        "humidity": 100,
        "temp_min": 7.78,
        "temp_max": 8.33
    },
    "visibility": 6000,
    "wind": {
        "speed": 3,
        "deg": 80
    },
    "rain": {
        "1h": 0.76
    },
    "clouds": {
        "all": 90
    },
    "dt": 1573154160,
    "sys": {
        "type": 1,
        "id": 8903,
        "country": "UA",
        "sunrise": 1573102677,
        "sunset": 1573136713
    },
    "timezone": 7200,
    "id": 703448,
    "name": "Kiev",
    "cod": 200
};