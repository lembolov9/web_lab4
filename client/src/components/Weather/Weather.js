import React from "react";
import "./Weather.css"

class Weather extends React.Component {

    componentDidMount() {
        this.props.getWeather();
    }

    render() {
        if (this.props.data.error) {
            return (
                <div className='Weather error'>
                    <div>Shit happens</div>
                    {this.props.data.id !== 0 && <button onClick={this.props.deleteCity} className='circle'>x</button>}
                </div>
            );
        } else {
            console.warn(this.props.data.id)
            const weather = this.props.data.weather;
            if (this.props.data.waiting === false) {
                const weather_img = (
                    this.props.data.id ? `https://openweathermap.org/img/wn/${weather.weather[0].icon}.png`:
                        `https://openweathermap.org/themes/openweathermap/assets/vendor/owm/img/widgets/${weather.weather[0].icon}.png`
                );
                return (
                    <div className={this.props.data.id ? 'Weather': 'Weather grid'}>
                        <div className={this.props.data.id ? 'mainInfo propsRow flex': 'mainInfo grid'}>
                            <div className="cityName">{this.props.data.id ? weather.name: <h3>{weather.name}</h3>}</div>
                            <div className="cityTemp">{weather.main.temp} â„ƒ</div>
                            <div className="cityIcon"><img alt="weather icon" src={weather_img}/></div>
                            {this.props.data.id !==0 && (<div><button onClick={this.props.deleteCity} className='deleteBtn'>+</button></div>) }
                        </div>
                        <div className="Info flex">
                            <div className='propsRow flex'>
                            <div>Wind speed:</div><div>{weather.wind.speed} m/s</div>
                            </div>
                            <div className='propsRow flex'>
                            <div>Clouds:</div><div>{weather.clouds.all}%</div>
                            </div>
                            <div className='propsRow flex'>
                            <div>Pressure:</div><div>{weather.main.pressure}hpa</div>
                            </div>
                            <div className='propsRow flex'>
                            <div>Humidity:</div><div>{weather.main.humidity}%</div>
                            </div>
                            <div className='propsRow flex'>
                            <div>Coordinates:</div><div>[{weather.coord.lon}, {weather.coord.lat}]</div>
                            </div>
                        </div>

                    </div>
                );
            } else {
                return (
                    <div className='Weather'>
                        <div className="loader"></div>
                    </div>
                );
            }
        }
    }
}
export default Weather;