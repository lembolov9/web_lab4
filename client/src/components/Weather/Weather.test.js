import React from "react";
import Weather from "./Weather";
import renderer from "react-test-renderer";
import correctWeather from "../../test-data/correctWeather";


describe("Main city weather", () => {
    it("Ready weather data", () => {
        const testData = {
            id: 0,
            waiting: false,
            weather: correctWeather
        }
        const component = renderer.create(<Weather data={testData} getWeather={()=> {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Not ready weather data", () => {
        const testData = {
            id: 0,
            waiting: true
        }
        const component = renderer.create(<Weather data={testData} getWeather={()=> {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Error", () => {
        const testData = {
            id: 0,
            waiting: false,
            error: "Upps"
        }
        const component = renderer.create(<Weather data={testData} getWeather={()=> {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

})

describe("Not main city weather", () => {
    it("Ready weather data", () => {
        const testData = {
            id: 100,
            waiting: false,
            weather: correctWeather
        }
        const component = renderer.create(<Weather data={testData} getWeather={()=> {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Not ready weather data", () => {
        const testData = {
            id: 100,
            waiting: true
        }
        const component = renderer.create(<Weather data={testData} getWeather={()=> {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Error", () => {
        const testData = {
            id: 100,
            waiting: false,
            error: "Upps"
        }
        const component = renderer.create(<Weather data={testData} getWeather={()=> {}}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

})

