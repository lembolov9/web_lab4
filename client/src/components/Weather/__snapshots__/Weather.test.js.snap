// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main city weather Error 1`] = `
<div
  className="Weather error"
>
  <div>
    Shit happens
  </div>
</div>
`;

exports[`Main city weather Not ready weather data 1`] = `
<div
  className="Weather"
>
  <div
    className="loader"
  />
</div>
`;

exports[`Main city weather Ready weather data 1`] = `
<div
  className="Weather grid"
>
  <div
    className="mainInfo grid"
  >
    <div
      className="cityName"
    >
      <h3>
        Kiev
      </h3>
    </div>
    <div
      className="cityTemp"
    >
      8.02
       ℃
    </div>
    <div
      className="cityIcon"
    >
      <img
        alt="weather icon"
        src="https://openweathermap.org/themes/openweathermap/assets/vendor/owm/img/widgets/10n.png"
      />
    </div>
  </div>
  <div
    className="Info flex"
  >
    <div>
      Wind speed:
    </div>
    <div>
      3
       m/s
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Clouds:
    </div>
    <div>
      90
      %
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Pressure:
    </div>
    <div>
      1011
      hpa
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Humidity:
    </div>
    <div>
      100
      %
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Coordinates:
    </div>
    <div>
      [
      30.52
      , 
      50.43
      ]
    </div>
  </div>
</div>
`;

exports[`Not main city weather Error 1`] = `
<div
  className="Weather error"
>
  <div>
    Shit happens
  </div>
  <button
    className="circle"
  >
    x
  </button>
</div>
`;

exports[`Not main city weather Not ready weather data 1`] = `
<div
  className="Weather"
>
  <div
    className="loader"
  />
</div>
`;

exports[`Not main city weather Ready weather data 1`] = `
<div
  className="Weather"
>
  <div
    className="mainInfo propsRow flex"
  >
    <div
      className="cityName"
    >
      Kiev
    </div>
    <div
      className="cityTemp"
    >
      8.02
       ℃
    </div>
    <div
      className="cityIcon"
    >
      <img
        alt="weather icon"
        src="https://openweathermap.org/img/wn/10n.png"
      />
    </div>
    <div>
      <button
        className="deleteBtn"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="Info flex"
  >
    <div>
      Wind speed:
    </div>
    <div>
      3
       m/s
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Clouds:
    </div>
    <div>
      90
      %
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Pressure:
    </div>
    <div>
      1011
      hpa
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Humidity:
    </div>
    <div>
      100
      %
    </div>
  </div>
  <div
    className="propsRow flex"
  >
    <div>
      Coordinates:
    </div>
    <div>
      [
      30.52
      , 
      50.43
      ]
    </div>
  </div>
</div>
`;
